packages/flame/lib/src/components/router_component.dart

```diff
@@ -123,6 +151,28 @@ class RouterComponent extends Component {
     _adjustRoutesVisibility();
   }
 
+  /// Pops the current route and puts a new [route] on top of the navigation
+  /// stack.
+  ///
+  /// The route may also be given a [name], in which case it will be cached in
+  /// the [routes] map under this name (if there was already a route with the
+  /// same name, it will be overwritten).
+  ///
+  /// The method calls [Route.didPush] for this new route after it is added and
+  /// also calls the [Route.didPop] callback for the popped route.
+  void pushReplacement(Route route, {String name = ''}) {
```

Comment:
`name` should probably be nullable instead of an empty string by default.

Reply:
I just copied the existing `pushRoute` to implement.  @st-pasha wrote that code originally, so I am not certain as to why it was implemented as an empty string, but the code calls:

```
 void pushRoute(Route route, {String name = ''}) {
    route.name = name;
    if (name.isNotEmpty) {
      _routes[name] = route;
    }
```
So it only pushes the name to the stack if its not empty.  I can go either way, as its the same outcome, but maybe he had a reason for it??

Reply:
Maybe he did, I can't come up with a good reason for using a string when it is not going to be used though, and theoretically an empty string is also a valid name for a route.

Reply:
I can change the code and run some tests to see if there is any impact.

Reply:
Ok, the answer is pretty easy, because in dart you can't have an optional named parameter of type string with a default value of null.  So you have to make it a required parameter which then causes other issues for unnamed routes.  Regardless, with it being an empty string, the if statement prevents the unnamed route from being added to the stack list, so you do not have multiple empty string route names or even one empty string route name.  Thus you can push an overlay or route that isn't named and once it is popped, it is destroyed.

Reply:
You can have optional parameters of default value null, it is the default without you having to add anything.
Check here:
https://zapp.run/edit/flame-z03w062x03x0?

Reply:
Ok, I must have seriously misread that then.  I have modified the code as necessary and all tests pass.  I will be curious what happens long term, because `route.dart` defines name like:

```
 /// The name of the route (set by the [RouterComponent]).
  String get name => _name;
  late String _name;
  @internal
  set name(String value) => _name = value;
  ```
  If its never set, which my code update prevents unless its not null, then is there anything that would ever call `route.name` and expect it to not be null when it could be because it was never set.  Regardless, I think its all working right.

Reply:
Ooh right, this was already in there, I missed that.
Let's see what @st-pasha says.

Reply:
I don't quite remember why I made the default name as the empty string. Maybe at first I thought the name must be required, and then when no-name routes were added later, it was the easiest way? Anyways, I don't mind if we make the name nullable. In which case the `name` getter should return `String?`, because we don't want it to throw an exception if the route has no name.

Reply:
This would be a breaking change then or no?

Reply:
It would, but quite minor, I don't think it would affect anyone.

Reply:
Ok.  I will update accordingly.  I'm at CES so it will be a day or two before I get to it.

Reply:
Cool, no rush!